apps/server/lib/lexical/server/provider/handlers/rename.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Lexical.Server.Provider.Handlers.Rename do
+  alias Lexical.Ast
+  alias Lexical.Document
+  alias Lexical.Protocol.Requests.Rename
+  alias Lexical.Protocol.Responses
+  alias Lexical.Protocol.Types.Workspace.Edit
+  alias Lexical.RemoteControl.Api
+  alias Lexical.Server.Provider.Env
+  require Logger
+
+  def handle(%Rename{} = request, %Env{} = env) do
+    case Document.Store.fetch(request.document.uri, :analysis) do
+      {:ok, _document, %Ast.Analysis{} = analysis} ->
```

Comment:
Analysis always succeeds, so we need to look at `valid?` on the analysis to actually get to the error case here.

```suggestion
      {:ok, _document, %Ast.Analysis{valid?: true} = analysis} ->
```